README.md

```diff
@@ -1,407 +1,150 @@
-- **Projection Support**: Perform selective conversion of CSV data subsets into Record values through projection.
+- **Versatile CSV Data Input**: Accept CSV data as a string, byte array, or a stream and convert it into a subtype of ballerina records or lists.
+- **CSV to anydata Value Conversion**: Transform CSV data into expected type which is subtype of ballerina records or lists.
+- **Projection Support**: Perform selective conversion of CSV data subsets into ballerina records or lists values through projection.
 
 ## Usage
 
-### Converting an CSV value to a Record value
+### Converting CSV string to a record array
 
-To convert an CSV value to a Record value, you can utilize the `fromCsvWithType` function provided by the library. The example below showcases the transformation of an CSV value into a Record value.
+To convert an CSV document value to a record value, you can utilize the `parseStringToRecord` function provided by the library. The example below showcases the transformation of a CSV document into a record array.
 
 ```ballerina
 import ballerina/data.csv;
 import ballerina/io;
 
-public function main() returns error? {
-    csv data = csv `<book>
-        <id>0</id>
-        <title>string</title>
-        <author>string</author>
-    </book>`;
+type Book record {
+    string name;
+    string author;
+    int year;
+};
 
-    Book book = check csvdata:fromCsvWithType(data);
-    io:println(book);
+public function main() returns error? {
+    string csvString = string `name,author,year
+                               Clean Code,Robert C. Martin,2008
+                               The Pragmatic Programmer,Andrew Hunt and David Thomas,1999`;
+
+    Book[] books = check csv:parseStringToRecord(csvString);
+    foreach var book in books {
+        io:println(book);
+    }
 }
-
-type Book record {|
-    int id;
-    string title;
-    string author;
-|};
 ```
 
-### Converting an external CSV document to a Record value
+### Converting external CSV document to a record value
 
-For transforming CSV content from an external source into a Record value, the `fromCsvStringWithType` function can be used. This external source can be in the form of a string or a byte array/byte stream that houses the CSV data. This is commonly extracted from files or network sockets. The example below demonstrates the conversion of an CSV value from an external source into a Record value.
+For transforming CSV content from an external source into a record value, the `parseStringToRecord`, `parseBytesToRecord`, `parseStreamToRecord`, `parseStringToList`, `parseBytesToList`and `parseStreamToList` functions can be used. This external source can be in the form of a string or a byte array/byte-block-stream that houses the CSV data. This is commonly extracted from files or network sockets. The example below demonstrates the conversion of an CSV value from an external source into a record value.
```

Comment:
I don't this we need this sub content. In here external simply means a reference.